theano/sandbox/gpuarray/extra_ops.py

```diff
@@ -0,0 +1,489 @@
+from __future__ import absolute_import, print_function, division
+import theano
+import numpy
+import os
+from theano import Op, Apply, config
+from theano.tensor.extra_ops import CumsumOp
+
+try:
+    import pygpu
+    from pygpu import gpuarray
+except ImportError:
+    pass
+
+from .basic_ops import (as_gpuarray_variable, GpuKernelBase, Kernel,
+                        infer_context_name, GpuFromHost, HideC)
+from .opt import register_opt as register_gpu_opt, op_lifter
+from .type import GpuArrayType
+
+
+class GpuCumsum(GpuKernelBase, HideC, CumsumOp):
+    """
+    Parameters
+    ----------
+    axis
+        Can not be None. If you want the array flattened, do it before.
+    """
+    SUPPORTED_NDIMS = 3
+    __props__ = ('axis',)
+
+    def __init__(self, axis):
+        self.axis = axis
+
+    def __str__(self):
+        return "%s{%s}" % (self.__class__.__name__, self.axis)
+
+    def c_code_cache_version_apply(self, node):
+        return None
```

Comment:
You probably want to turn that back into `c_code_cache_version` and make it return some value.